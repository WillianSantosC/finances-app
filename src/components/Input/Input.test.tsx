import { fireEvent, render } from '@testing-library/react-native';
import { Input } from '.';

describe('<Input />', () => {
  it('should render label correctly', () => {
    const { getByText } = render(<Input label="E-mail" />);
    const label = getByText('E-mail');
    expect(label).toBeTruthy();
  });

  it('should render placeholder correctly', () => {
    const { getByPlaceholderText } = render(<Input placeholder="Digite seu e-mail" />);
    const input = getByPlaceholderText('Digite seu e-mail');
    expect(input).toBeTruthy();
  });

  it('should display error message when provided', () => {
    const { getByText } = render(<Input errorMessage="Campo obrigatório" />);
    const error = getByText('Campo obrigatório');
    expect(error).toBeTruthy();
  });

  it('should call onChangeText correctly', () => {
    const onChangeTextMock = jest.fn();

    const { getByPlaceholderText } = render(
      <Input placeholder="Digite algo" onChangeText={onChangeTextMock} />
    );

    const input = getByPlaceholderText('Digite algo');

    fireEvent.changeText(input, 'Testando');

    expect(onChangeTextMock).toHaveBeenCalledWith('Testando');
  });

  it('should match snapshot without error', () => {
    const tree = render(<Input label="Senha" placeholder="Digite sua senha" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with error', () => {
    const tree = render(
      <Input label="Senha" placeholder="Digite sua senha" errorMessage="Senha inválida" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
