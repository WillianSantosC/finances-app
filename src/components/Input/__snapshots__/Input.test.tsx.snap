// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot with error 1`] = `
<View
  className="w-[90%] mb-4"
>
  <Text
    className="mb-1 text-base text-black font-medium"
  >
    Senha
  </Text>
  <View
    className="flex-row items-center rounded-lg bg-white px-3 border border-red"
  >
    <TextInput
      className="flex-1 h-[45px] text-[17px] text-black"
      placeholder="Digite sua senha"
      placeholderTextColor="#A1A1A1"
    />
  </View>
  <Text
    className="mt-1 text-sm text-red"
  >
    Senha inv√°lida
  </Text>
</View>
`;

exports[`<Input /> should match snapshot without error 1`] = `
<View
  className="w-[90%] mb-4"
>
  <Text
    className="mb-1 text-base text-black font-medium"
  >
    Senha
  </Text>
  <View
    className="flex-row items-center rounded-lg bg-white px-3 border border-[#E8E8E8]"
  >
    <TextInput
      className="flex-1 h-[45px] text-[17px] text-black"
      placeholder="Digite sua senha"
      placeholderTextColor="#A1A1A1"
    />
  </View>
</View>
`;
